Tools/autotest/common.py

```diff
@@ -846,26 +839,32 @@ def disarm_motors_with_switch(self, switch_chan, timeout=20):
             if not self.mav.motors_armed():
                 self.progress("MOTORS DISARMED OK WITH SWITCH")
                 return True
-        self.progress("FAILED TO DISARM WITH SWITCH")
+        self.progress("Unable to DISARM with switch")
         return False
 
-    def autodisarm_motors(self):
-        """Autodisarm motors."""
-        self.progress("Autodisarming motors")
-        if self.mav.mav_type == mavutil.mavlink.MAV_TYPE_GROUND_ROVER:  # NOT IMPLEMENTED ON ROVER
-            self.progress("MOTORS AUTODISARMED OK")
-            return True
+    def wait_autodisarm_motors(self):
+        """Wait for Autodisarm motors within disarm delay
+        this feature is only available in copter (DISARM_DELAY) and plane (LAND_DISARMDELAY)."""
+        self.progress("Wait autodisarming motors")
+        disarm_delay = self.get_disarm_delay()
         tstart = self.get_sim_time()
-        timeout = 15
+        timeout = disarm_delay * 2
         while self.get_sim_time() < tstart + timeout:
             self.wait_heartbeat()
             if not self.mav.motors_armed():
-                disarm_delay = self.get_sim_time() - tstart
+                disarm_time = self.get_sim_time() - tstart
                 self.progress("MOTORS AUTODISARMED")
-                self.progress("Autodisarm in %ss" % disarm_delay)  # TODO check disarming time
-                return True
-        self.progress("FAILED TO AUTODISARM")
-        return False
+                self.progress("Autodisarm in %ss, expect less than %ss" % (disarm_time, disarm_delay))
+                return disarm_time <= disarm_delay
+        self.progress("Unable to AUTODISARM")
+        raise AutoTestTimeoutException()
+
+    def set_autodisarm_delay(self, delay):
+        """Set autodisarm delay"""
```

Comment:
Implement set_disarm_delay on subclasses, fatal in parent class to be consistent with getters.